code/search/SearchQuery.php

```diff
@@ -77,32 +100,116 @@ function filter($field, $values) {
 	 * @param  String $field
 	 * @param  mixed $values
 	 */
-	function exclude($field, $values) {
+	function addExclude($field, $values) {
 		$excludes = isset($this->exclude[$field]) ? $this->exclude[$field] : array();
 		$values = is_array($values) ? $values : array($values);
 		$this->exclude[$field] = array_merge($excludes, $values);
+		return $this;
 	}
 
-	function start($start) {
+	function getExcludes() {
+		return $this->exclude;
+	}
+
+	function setStart($start) {
 		$this->start = $start;
+		return $this;
 	}
 
-	function limit($limit) {
+	function getStart() {
+		return $this->start;
+	}
+
+	function setLimit($limit) {
 		$this->limit = $limit;
+		return $this;
 	}
 
-	function page($page) {
+	function getLimit() {
+		return $this->limit;
+	}
+
+	function setPageSize($page) {
```

Comment:
I think these methods need `public ` and some phpdoc. I'm not sure what this is doing immediately by reading the code.